presto-hive/src/main/java/com/facebook/presto/hive/ManifestPartitionLoader.java

```diff
@@ -0,0 +1,218 @@
+import static com.facebook.presto.hive.HiveManifestUtils.decompressFileSizes;
+import static com.facebook.presto.hive.HiveSessionProperties.getMaxInitialSplitSize;
+import static com.facebook.presto.hive.HiveSessionProperties.getMaxSplitSize;
+import static com.facebook.presto.hive.HiveSessionProperties.getNodeSelectionStrategy;
+import static com.facebook.presto.hive.HiveSessionProperties.isManifestVerificationEnabled;
+import static com.facebook.presto.hive.HiveUtil.getInputFormat;
+import static com.facebook.presto.hive.NestedDirectoryPolicy.IGNORED;
+import static com.facebook.presto.hive.NestedDirectoryPolicy.RECURSE;
+import static com.facebook.presto.hive.metastore.MetastoreUtil.getPartitionLocation;
+import static com.google.common.base.Verify.verify;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+public class ManifestPartitionLoader
+        extends PartitionLoader
+{
+    private final Table table;
+    private final Optional<Domain> pathDomain;
+    private final ConnectorSession session;
+    private final HdfsEnvironment hdfsEnvironment;
+    private final HdfsContext hdfsContext;
+    private final NamenodeStats namenodeStats;
+    private final DirectoryLister directoryLister;
+    private final boolean recursiveDirWalkerEnabled;
+    private final boolean schedulerUsesHostAddresses;
+
+    public ManifestPartitionLoader(
+            Table table,
+            Optional<Domain> pathDomain,
+            ConnectorSession session,
+            HdfsEnvironment hdfsEnvironment,
+            NamenodeStats namenodeStats,
+            DirectoryLister directoryLister,
+            boolean recursiveDirWalkerEnabled,
+            boolean schedulerUsesHostAddresses)
+    {
+        this.table = requireNonNull(table, "table is null");
+        this.pathDomain = requireNonNull(pathDomain, "pathDomain is null");
+        this.session = requireNonNull(session, "session is null");
+        this.hdfsEnvironment = requireNonNull(hdfsEnvironment, "hdfsEnvironment is null");
+        this.hdfsContext = new HdfsContext(session, table.getDatabaseName(), table.getTableName());
+        this.namenodeStats = requireNonNull(namenodeStats, "namenodeStats is null");
+        this.directoryLister = requireNonNull(directoryLister, "directoryLister is null");
+        this.recursiveDirWalkerEnabled = recursiveDirWalkerEnabled;
+        this.schedulerUsesHostAddresses = schedulerUsesHostAddresses;
+    }
+
+    public ListenableFuture<?> loadPartition(HivePartitionMetadata partition, HiveSplitSource hiveSplitSource, boolean stopped)
+            throws IOException
+    {
+        Path path = new Path(getPartitionLocation(table, partition.getPartition()));
+        Map<String, String> parameters = partition.getPartition().get().getParameters();
+        List<String> fileNames = decompressFileNames(parameters.get(FILE_NAMES));
```

Comment:
- Also get manifest version and validate it's "V1"
- Add a TODO to make versioning available in the future